### ISSUE TITLE
PARALLEL: Make use of futures

### ISSUE THREAD

**Issue by HenrikBengtsson** [HenrikBengtsson/PSCBS#22]
Make use of futures when processing for instance data chromosome by chromosome.  Many functions can be updated, notably:
* [x] `segmentByCBS()`
* [x] `segmentByPairedPSCBS()`

**Comment by HenrikBengtsson** [comment]
When only one chromosome is processed, should we force to use "eager" futures?

**Comment by HenrikBengtsson** [comment]
This is already done, that is, futures are only used if `length(chromosomes) > 1`, e.g. https://github.com/HenrikBengtsson/PSCBS/blob/master/R/segmentByCBS.R#L334-L335.

**Comment by HenrikBengtsson** [comment]
For anyone interested, there's now an early access version available where `segmentByCBS()` and `segmentByPairedPSCBS()` can segment the chromosomes in parallel using [futures](https://cran.r-project.org/package=future).  All that is need to add `future::plan("multicore")` on top of script.  Alternatively, one can set environment variable `R_FUTURE_PLAN=multicore`. To try this out, see:

https://github.com/HenrikBengtsson/PSCBS/tree/feature/future#pre-release-version

(Make sure to update to the latest `future` and `globals` versions now available on CRAN)

CC: @olshena, @pneuvial, @mpierrejean, @etal

**Comment by HenrikBengtsson** [comment]
@olshena, @veseshan, @pneuvial, @mpierrejean, @etal - FYI,

PSCBS 0.60.0 is now on [CRAN](https://cran.r-project.org/web/packages/PSCBS) and supports parallel segmentation, i.e. each chromosome can be segmented in parallel.  To enable, see https://github.com/HenrikBengtsson/PSCBS/.

Extra [care was taken](#30) to make sure we get numerical reproducibility regardless on whether synchronous or asynchronous processing was used and exactly what parallel/distributed backend is used.  [Several](https://github.com/HenrikBengtsson/PSCBS/blob/master/tests/segmentByCBS%2Cfutures.R) [package](https://github.com/HenrikBengtsson/PSCBS/blob/master/tests/segmentByPairedPSCBS%2Cfutures.R) [tests](https://github.com/HenrikBengtsson/PSCBS/blob/master/tests/randomSeed.R) were extended to assert this.

Hopefully you won't run into problems/surprises.  Feedback is appreciated (post a [new issue](https://github.com/HenrikBengtsson/PSCBS/issues/new) if you run into problems).
